src/com/sun/jna/Function.java

```diff
@@ -245,7 +245,7 @@ public static Function getFunction(Pointer p, int callFlags, String encoding) {
         this.library = library;
         this.functionName = functionName;
         this.callFlags = callFlags;
-        this.options = library.options;
+        this.options = library.getOptions();
```

Comment:
This map is still mutable; if someone modified the library map (e.g. `getOptions.clear()` that would echo here.  

Now that we are on JDK8 I'd suggest making this version of it an unmodifiable copy.  (I'd suggest making the getter return unmodifiable but that's probably a breaking change.)
```suggestion
        this.options = Map.copyOf(library.getOptions());
```